PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ServiceBusReceiveModeCheck.java

```diff
@@ -0,0 +1,241 @@
+            if (elements.length > 0 && elements[0] instanceof PsiVariable) {
+                PsiVariable variable = (PsiVariable) elements[0];
+
+                // Process the variable declaration
+                processVariableDeclaration(variable);
+            }
+        }
+
+        /**
+         * This method is used to process the variable declaration.
+         * It checks if the client type is an Azure ServiceBus client and
+         * retrieves the client name (left side of the declaration).
+         *
+         * @param variable The variable to process
+         */
+        private void processVariableDeclaration(PsiVariable variable) {
+
+            // Retrieve the client name (left side of the declaration)
+            PsiType clientType = variable.getType();
+
+            // Check if the client type is an Azure ServiceBus client
+            if (!clientType.getCanonicalText().startsWith(RuleConfig.AZURE_PACKAGE_NAME) && !CLIENTS_TO_CHECK.contains(clientType.getCanonicalText())) {
+                return;
+            }
+
+            // Check the assignment part (right side)
+            PsiExpression initializer = variable.getInitializer();
+
+            if (!(initializer instanceof PsiMethodCallExpression)) {
+                return;
+            }
+
+            // Process the new expression initialization
+            determineReceiveMode((PsiMethodCallExpression) initializer);
+
+        }
+
+        /**
+         * This method is used to determine the receive mode of the Azure ServiceBus client.
+         * It checks if the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1.
+         * If the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1,
+         * a problem is registered with the ProblemsHolder.
+         *
+         * @param methodCall The method call expression to check
+         */
+        private void determineReceiveMode(PsiMethodCallExpression methodCall) {
+
+            OptionalInt prefetchCountValue = OptionalInt.empty();
+            boolean isReceiveModePeekLock = false;
+            PsiElement prefetchCountMethod = null;
+
+            // Iterating up the chain of method calls
+            PsiExpression qualifier = methodCall.getMethodExpression().getQualifierExpression();
+
+            // Check if the method call chain has the method to check
+            while (qualifier instanceof PsiMethodCallExpression) {
+
+                // Get the method expression
+                PsiReferenceExpression methodExpression = ((PsiMethodCallExpression) qualifier).getMethodExpression();
+
+                // Get the method name
+                String methodName = methodExpression.getReferenceName();
+
+                // Check if the method name is the method to check
+                // "receiveMode is the first method to check"
+
+                if (METHODS_TO_CHECK.get(0).equals(methodName)) {
```

Comment:
This logic is a fragile because what if they reorder the elements in your config file? You also don't have any logic in the static constructor to guarantee that there are 2 values in this array.

Reply:
I was trying to avoid using magic strings but I think the alternative is more complicated and I only use them once so it should be fine to use them